spring-integration-event/src/main/java/org/springframework/integration/event/inbound/ApplicationEventListeningMessageProducer.java

```diff
@@ -116,20 +129,45 @@ public void onApplicationEvent(ApplicationEvent event) {
 		}
 	}
 
+	@Override
+	protected Object evaluatePayloadExpression(Object rootObject) {
+		if (!this.payloadExpressionSet && rootObject instanceof PayloadApplicationEvent) {
+			return ((PayloadApplicationEvent) rootObject).getPayload();
```

Comment:
I see your point about `@EventListener`, but I still don't know why we'd evaluate an expression against the event.